### ISSUE TITLE
Clarify AnimationOptions#offset documentation – move to CameraOptions

### ISSUE THREAD

**Issue by mollymerp** [mapbox/mapbox-gl-js#4240]
We currently have an existing solution to the top requested feature ( #1740 ) but it's not clear that we do because the docs on this option are confusing. 

It is currently in `AnimationOptions` described as:
> x and y coordinates representing the animation's origin of movement relative to the map's center.

I think it should be moved to `CameraOptions` because it affects the Map's positioning after an animation completes, and the documentation should be reworded so it is understandable by regular humans.

**Comment by ansis** [comment]
I agree that `AnimationOptions` is a weird place for this but I don't think we can change it anymore. https://github.com/mapbox/mapbox-gl-js/issues/1740 has been fixed with `PaddingOptions`.

We still need to improve the doc for this property.
